### PR TITLE
Move to Project.toml and update Travis

### PR DESCRIPTION
**Pull request by IanButterworth**

### CODE REVIEW

**Comment by IanButterworth** [comment]
I can't get my head around the Unsatisfiable requirements error.. It doesn't happen on my local MacOS machine on 1.1
```
ERROR: Unsatisfiable requirements detected for package ImageCore [a09fc81d]:
 ImageCore [a09fc81d] log:
 ├─possible versions are: [0.0.1-0.0.5, 0.1.0-0.1.5, 0.2.0-0.2.1, 0.3.0-0.3.3, 0.4.0-0.4.1, 0.5.0, 0.6.0, 0.7.0-0.7.4, 0.8.0] or uninstalled
 ├─restricted to versions * by ImageMagick [6218d12a], leaving only versions [0.0.1-0.0.5, 0.1.0-0.1.5, 0.2.0-0.2.1, 0.3.0-0.3.3, 0.4.0-0.4.1, 0.5.0, 0.6.0, 0.7.0-0.7.4, 0.8.0]
 │ └─ImageMagick [6218d12a] log:
 │   ├─possible versions are: 0.7.2 or uninstalled
 │   └─ImageMagick [6218d12a] is fixed to version 0.7.2
 ├─restricted to versions 0.8.0 by an explicit requirement, leaving only versions 0.8.0
 └─restricted by compatibility requirements with ImageShow [4e3cecfd] to versions: [0.0.1-0.0.5, 0.1.0-0.1.5, 0.2.0-0.2.1, 0.3.0-0.3.3, 0.4.0-0.4.1, 0.5.0, 0.6.0, 0.7.0-0.7.4] — no versions left
   └─ImageShow [4e3cecfd] log:
     ├─possible versions are: 0.1.0-0.1.2 or uninstalled
     └─restricted to versions * by an explicit requirement, leaving only versions 0.1.0-0.1.2
```

File: Project.toml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+InteractiveUtils = "b77e0a4c-d291-57a0-90e8-8db25a27a240"
+Libdl = "8f399da3-3557-5675-b5ff-fb832c97cbdb"
+Pkg = "44cfe95a-1eb2-52ea-b672-e2afdf69b78f"
+Random = "9a3f8284-a2c9-5f02-9a11-845980a1fd5c"
+
+[compat]
+BinaryProvider = "≥ 0.3.0"
+ColorTypes = "≥ 0.2.7"
+FixedPointNumbers = "≥ 0.3.0"
+julia = "≥ 0.7.0"
```
Comment (johnnychen94) [inline Project.toml]: Perhaps we can drop 0.7 compatibility since we drop the CI test.

**Comment by johnnychen94** [comment]
This PR is blocked by https://github.com/JuliaImages/ImageShow.jl/pull/8, retrigger the CI when  ImageShow gets a version bump.

ping @evizero @timholy

**Comment by IanButterworth** [comment]
It would be great to resolve this. I'm trying to fix some issues in VideoIO that are being held up by the deadlock between this and JuliaImages/ImageShow.jl#8 

Can we go ahead and merge and version this before CI passes? Or perhaps is there a way to temporarily force CI to check out the `johnnychen94:new-registrator` branch of `ImageShow` to prove it works before either is versioned?

**Comment by johnnychen94** [comment]
Since `ImageShow` is test-stage dependency, it won't affect the actual functionality of `ImageMagick`. Add compatibility restriction to `ImageShow` seems reasonable.

**Comment by IanButterworth** [comment]
@johnnychen94 I have to admit, I can't figure out what compatibility restriction to enable for `ImageShow`... Have you figured it out? I actually still don't understand why this is happening

File: Project.toml (hunk 1 above)
Comment (IanButterworth) [reply Project.toml]: Done 👍🏻

**Comment by IanButterworth** [comment]
So I realized you meant `ImageCore = "< 0.8"` and I've implemented that and fixed a Travis setup bug. 
Build is succeeding now.

I think a lot of my confusion is that I didn't understand why this was happening because I didn't realize that Pkg3 doesn't downgrade any package thats only a dep of a dep https://github.com/JuliaLang/Pkg.jl/issues/110

**Comment by IanButterworth** [comment]
@johnnychen94 I've requested installation of the Registrator onto this repo and will version when that happens

**Pull request merged**